src/main/java/com/webperside/courseerpbackend/repository/TeacherRepository.java

```diff
@@ -0,0 +1,18 @@
+package com.webperside.courseerpbackend.repository;
+
+import com.webperside.courseerpbackend.models.mybatis.teacher.Teacher;
+import org.apache.ibatis.annotations.Mapper;
+
+import java.util.List;
+import java.util.Optional;
+
+@Mapper
+public interface TeacherRepository {
+    void insert(Teacher teacher);
+
+    void update(Teacher teacher);
+
+    Optional<Teacher> findById(Long id);
```

Comment:
Inside the findById to be good to use @Param("id") annotation;

Reply:
actually it is not required to use @Param. my batis will use field name by default, if you don't specify a name.

Reply:
but it is recommended :)